src/cljs/pisano_cx_glossary/util.cljs

```diff
@@ -2,46 +2,36 @@
   (:require [ajax.core :as ajax]
             [clojure.string :as str]))
 
-(defn create-custom-request-map
-  ([type uri]
-   (create-custom-request-map type uri nil nil))
-  ([type uri on-success]
-   (create-custom-request-map type uri on-success nil))
-  ([type uri on-success on-fail]
-   (cond-> {:method          type
-            :uri             uri
-            :format          (ajax/json-request-format)
-            :response-format (ajax/json-response-format {:keywords? true})
-            :on-success      (if (vector? on-success) on-success [on-success])
-            :on-failure      (if (vector? on-fail) on-fail [on-fail])}
-           (nil? on-success) (assoc :on-success [:no-http-on-ok])
-           (nil? on-fail) (assoc :on-failure [:no-http-on-failure]))))
+(def x->X {"ı" "I" 
+           "i" "İ"
+           "ç" "Ç"
+           "ö" "Ö"
+           "ü" "Ü"
+           "ğ" "Ğ"
+           "ş" "Ş"})
 
 
 (defn upper-case
   [s]
-  (some-> s
-          (str/replace #"ı" "I")
-          (str/replace #"i" "İ")
-          (str/replace #"ç" "Ç")
-          (str/replace #"ö" "Ö")
-          (str/replace #"ü" "Ü")
-          (str/replace #"ğ" "Ğ")
-          (str/replace #"ş" "Ş")
-          (str/upper-case)))
+  (when s
+    (reduce-kv (fn [acc x X] (str/replace acc x X)) (str/upper-case s) x->X)))
```

Comment:
TODO: check if `(defn upper-case [s] (when s (str/upper-case s)))` works well with Turkish letters

Reply:
Nope it does not :)